src/org/traccar/reports/ReportUtils.java

```diff
@@ -250,100 +255,79 @@ private static boolean isMoving(ArrayList<Position> positions, int index,
-                                && tripDistance < tripsConfig.getMinimalTripDistance();
+                if (deviceState.getMotionState()) {
+                    if (startParkingIndex == -1) {
                         startParkingIndex = i;
+                    } else if (deviceState.getMotionPosition() == null) {
+                        startParkingIndex = -1;
                     }
                 }
-                if (startParkingIndex != -1 && (endParkingIndex > startParkingIndex || isLast)) {
-                    long parkingDuration = positions.get(endParkingIndex).getFixTime().getTime()
-                            - positions.get(startParkingIndex).getFixTime().getTime();
-                    if ((parkingDuration >= tripsConfig.getMinimalParkingDuration() || isLast)
-                            && previousEndParkingIndex < startParkingIndex) {
-                        if (!tripFiltered) {
-                            if (trips) {
-                                result.add(calculateTrip(
-                                        positions, previousEndParkingIndex, startParkingIndex, ignoreOdometer));
-                            } else {
-                                if (result.isEmpty() && previousEndParkingIndex > previousStartParkingIndex) {
-                                    long previousParkingDuration = positions.get(previousEndParkingIndex)
-                                            .getFixTime().getTime() - positions.get(previousStartParkingIndex)
-                                            .getFixTime().getTime();
-                                    if (previousParkingDuration >= tripsConfig.getMinimalParkingDuration()) {
-                                        result.add(calculateStop(positions, previousStartParkingIndex,
-                                                previousEndParkingIndex));
-                                    }
-                                }
-                                result.add(calculateStop(positions, startParkingIndex, isLast ? i : endParkingIndex));
-                            }
-                        }
-                        previousEndParkingIndex = endParkingIndex;
-                        skipped = false;
-                    } else {
-                        skipped = true;
+                if (startTripIndex != -1 && startParkingIndex != -1 && event != null && !deviceState.getMotionState()) {
+                    result.add(calculateTrip(positions, startTripIndex, startParkingIndex, ignoreOdometer));
+                    startTripIndex = -1;
+                }
+            }
+            if (startTripIndex != -1 && startParkingIndex != -1) {
+                result.add(calculateTrip(positions, startTripIndex, startParkingIndex, ignoreOdometer));
+            }
+        }
+        return result;
+    }
+
+    public static Collection<StopReport> detectStops(Collection<Position> positionCollection, TripsConfig tripsConfig,
```

Comment:
Detect trips and detect stops logic is very similar, but basically inverse. Is there a way we can combine two?